typedargs/typeinfo.py

```diff
@@ -235,6 +237,26 @@ def _is_factory(cls, typeobj):
 
         return False
 
+    @staticmethod
+    def get_type_name(type_class: type) -> Union[str, None]:
```

Comment:
This function should not be necessary.

There's a larger refactor that we want to do in order to eventually get rid of the need for `type_name` altogether.

Right now, `type_name` is needed because it's used to look up the type inside a dictionary of known types in the `type_system`.

However, if you have the class itself, you don't need to do that since you already have the type.

So, the ultimate goal should be that `type_name` itself is optional and only needed for supporting legacy code that does not pass types in via function annotations.

See this diagram:
![image](https://user-images.githubusercontent.com/2424759/74115188-c576d900-4b62-11ea-993d-e55b0861da3c.png)